### DISCUSSION TITLE
setup.py is not available in Netmiko in Github

### DISCUSSION THREAD

**Discussion by gajigune** [ktbyers/netmiko#3516] [Q&A]
I am using the git clone command to insytall netmiko in Ubuntu 24.04. Command used below
$ git clone https://github.com/ktbyers/netmiko.git netmiko

$ ls
Desktop    Downloads  mytest_env  netmiko   Pictures  snap      Templates
Documents  Music      napalm      paramiko  Public    Students  Videos
$ cd netmiko

ls -l
total 388
-rw-rw-r--  1 gayathri gayathri   3634 ඔක්   27 21:50 COMMON_ISSUES.md
-rw-rw-r--  1 gayathri gayathri     26 ඔක්   27 21:50 _config.yml
-rw-rw-r--  1 gayathri gayathri   4644 ඔක්   27 21:50 CONTRIBUTING.md
drwxrwxr-x  3 gayathri gayathri   4096 ඔක්   27 21:50 docs
drwxrwxr-x  2 gayathri gayathri   4096 ඔක්   27 21:50 examples
-rw-rw-r--  1 gayathri gayathri  25717 ඔක්   27 21:50 EXAMPLES.md
drwxrwxr-x  8 gayathri gayathri   4096 ඔක්   27 21:50 examples_old
drwxrwxr-x  2 gayathri gayathri   4096 ඔක්   27 21:50 images
-rw-rw-r--  1 gayathri gayathri   1078 ඔක්   27 21:50 LICENSE
-rw-rw-r--  1 gayathri gayathri    422 ඔක්   27 21:50 license-dependencies.txt
-rw-rw-r--  1 gayathri gayathri     82 ඔක්   27 21:50 MANIFEST.in
drwxrwxr-x 78 gayathri gayathri   4096 ඔක්   27 23:11 netmiko
-rw-rw-r--  1 gayathri gayathri   4970 ඔක්   27 21:50 PLATFORMS.md
-rw-rw-r--  1 gayathri gayathri 262061 ඔක්   27 21:50 poetry.lock
-rw-rw-r--  1 gayathri gayathri   1798 ඔක්   27 21:50 pyproject.toml
-rw-rw-r--  1 gayathri gayathri   6498 ඔක්   27 21:50 README.md
-rw-rw-r--  1 gayathri gayathri    782 ඔක්   27 21:50 release_process.txt
-rwxrwxr-x  1 gayathri gayathri   3728 ඔක්   27 21:50 _release.sh
-rw-rw-r--  1 gayathri gayathri   1418 ඔක්   27 21:50 setup.cfg
-rw-rw-r--  1 gayathri gayathri    997 ඔක්   27 21:50 TESTING.md
-rw-rw-r--  1 gayathri gayathri   6465 ඔක්   27 21:50 test.log
drwxrwxr-x 19 gayathri gayathri   4096 ඔක්   27 21:50 tests
-rwxrwxr-x  1 gayathri gayathri    306 ඔක්   27 21:50 tests.sh
-rw-rw-r--  1 gayathri gayathri   2300 ඔක්   27 21:50 VENDOR.md

setup.py is unavailable

 python3 setup.py install
python3: can't open file '/home/gayathri/netmiko/setup.py': [Errno 2] No such file or directory

Please help

GG

**Comment by ktbyers** [comment]
Netmiko uses Python poetry and consequently doesn't have a setup.py file.

**Reply by gajigune** [reply]
Thank you. Please guide how to install netmiko in Ubuntu with Python poetry
try

**Reply by ktbyers** [reply]
Is there some reason you are not just executing "pip install netmiko"?

But for Poetry just look up Poetry documentation, it is pretty good.
